project/cmake/modules/FindBluetooth.cmake

```diff
@@ -0,0 +1,29 @@
+# Try to find Bluetooth
+#  BLUETOOTH_FOUND - If false, do not try to use Bluetooth.
+#  BLUETOOTH_INCLUDE_DIRS - where to find bluetooth/bluetooth.h
+#  BLUETOOTH_LIBRARIES - the libraries to link against
+
+if(BLUETOOTH_LIBRARIES AND BLUETOOTH_INCLUDE_DIRS)
+  set(BLUETOOTH_FOUND TRUE)
+else()
```

Comment:
Ah, I guess you check them because of the variable caching:

The _DIRS and _LIBRARIES vars should not be cached (which find_* does). Cached vars won't ever be updated, so if you have to add another folder or lib to them, you have to remove CMakeCache.txt. It's better to use find_path/find_libraries with different var (convention: _INCLUDE_DIR and _LIBRARY). And also mark those as advanced then.

Any reason for not using FindPackageHandleStandardArgs? That takes care of REQUIRED, QUIETLY and sets _FOUND.

You can use https://github.com/xbmc/xbmc/blob/556636b0c9a91856d05a02c22ef4ad0836610747/project/cmake/modules/FindVDPAU.cmake as a reference for find modules. The documentation has some more details: https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#find-modules